libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsTranscriptStore.java

```diff
@@ -0,0 +1,423 @@
+package com.microsoft.bot.azure.blobs;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import com.azure.storage.blob.models.BlobItem;
+import com.microsoft.bot.builder.BotAssert;
+import com.microsoft.bot.builder.PagedResult;
+import com.microsoft.bot.builder.TranscriptInfo;
+import com.microsoft.bot.builder.TranscriptStore;
+import com.microsoft.bot.restclient.serializer.JacksonAdapter;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.ActivityTypes;
+import com.microsoft.bot.schema.ChannelAccount;
+import com.microsoft.bot.schema.Pair;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.joda.time.DateTime;
+
+import javax.annotation.Nullable;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * The blobs transcript store stores transcripts in an Azure Blob container.
+ * Each activity is stored as json blob in structure of
+ * container/{channelId]/{conversationId}/{Timestamp.ticks}-{activity.id}.json.
+ */
+public class BlobsTranscriptStore implements TranscriptStore {
+
+    // Containers checked for creation.
+    private static HashSet<String> checkedContainers = new HashSet<String>();
+
+    // If a JsonSerializer is not provided during construction, this will be the default static JsonSerializer.
+    private final JacksonAdapter jacksonAdapter;
+    private BlobContainerClient containerClient;
+
+    public BlobsTranscriptStore(String dataConnectionString, String containerName) {
+        if (StringUtils.isBlank(dataConnectionString)) {
+            throw new IllegalArgumentException("dataConnectionString");
+        }
+
+        if (StringUtils.isBlank(containerName)) {
+            throw new IllegalArgumentException("containerName");
+        }
+
+        jacksonAdapter = new JacksonAdapter();
```

Comment:
This should be removed. Also, have you researched about the configuration that we can set in the [JacksonAdapter](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsTranscriptStore.cs#L58)?

Reply:
Also, I think that this should be replaced by ObjectMapper instead of jacksonAdapter in order to follow the approach of [CosmosDbPartitionedStorage](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-azure/src/main/java/com/microsoft/bot/azure/CosmosDbPartitionedStorage.java#L103)